components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/UpdateAuthenticationSequence.java

```diff
@@ -188,6 +184,25 @@ private AuthenticationStep buildAuthenticationStep(AuthenticationStepModel stepM
         return authenticationStep;
     }
 
+    private DefinedByType resolveDefinedByType(Authenticator authenticator) {
+
+        if (FrameworkConstants.LOCAL_IDP_NAME.equals(authenticator.getIdp())) {
+            LocalAuthenticatorConfig localAuthConfig = ApplicationAuthenticatorService.getInstance()
+                    .getLocalAuthenticatorByName(authenticator.getAuthenticator());
+            if (localAuthConfig != null) {
+                return localAuthConfig.getDefinedByType();
+            }
+        } else {
+            FederatedAuthenticatorConfig federatedAuthConfig = ApplicationAuthenticatorService.getInstance()
+                    .getFederatedAuthenticatorByName(authenticator.getAuthenticator());
+            if (federatedAuthConfig != null) {
+                return federatedAuthConfig.getDefinedByType();
+            }
+        }
+
+        return DefinedByType.USER;
```

Comment:
Why does the default becomes 'USER' ?

Reply:
First trying to retrieve from the system defined authenticator and if not found, returen USER.